mno_analysis_tools/compute_msg_difference_btwn_periods.py

```diff
@@ -0,0 +1,114 @@
+        input = json.load(f)
+        messages = [Message.deserialize(val) for val in input]
+        log.info(f"Loaded {len(messages)} messages")
+
+    # Filter messages based on the target operator and target direction of the message
+    log.info(f"Filtering messages based on operator {target_operator} and "
+             "message direction {target_message_direction} from {len(messages)} total messages ")
+    filtered_messages = []
+    for msg in messages:
+        if msg.urn.startswith("tel:"):
+            operator = PhoneCleaner.clean_operator(msg.urn.split(":")[1])
+        else:
+            operator = msg.urn.split(":")[0]
+        if operator == target_operator and msg.direction == target_message_direction:
+            msg_direction = msg.direction
+            filtered_messages.append(msg)
+    log.info(
+        f" returning {len(filtered_messages)} {target_operator} {target_message_direction} messages")
+
+    time_interval = timedelta(hours=time_frame.hour,
+                              minutes=time_frame.minute, seconds=time_frame.second)
+
+    date_time_bounds = date_time_range(start_date, end_date, time_interval)
+
+    # Compute number of messages between two datetime bounds i.e `PreviousMessageTimestamp` and
+    # `NextMessageTimestamp` to get number of mesages per period and relate each quantity
+    #  with the operator and the message direction.
+    messages_per_period = []
+    for index in range(len(date_time_bounds) - 1):
+        next_index = index + 1
+
+        messages_this_period = 0
+        for msg in filtered_messages:
+            if date_time_bounds[index] <= msg.sent_on < date_time_bounds[next_index]:
+                messages_this_period += 1
+
+        messages_per_period.append({
+            "Operator": operator,
+            "MessageDirection": msg_direction,
+            "PeriodStart": date_time_bounds[index].isoformat(),
+            "PeriodEnd": date_time_bounds[next_index].isoformat(),
+            "NumberOfMessages": messages_this_period
+        })
+
+    # Compute message difference between two firebase time periods
+    message_difference_per_period = []
+    for index in range(len(messages_per_period) - 1):
+        next_index = index + 1
+        message_difference_per_period.append({
+            "Operator": operator,
+            "MessageDirection": msg_direction,
+            "PeriodStart": messages_per_period[index]["PeriodStart"],
+            "PeriodBetween": messages_per_period[index]["PeriodEnd"],
+            "PeriodEnd": messages_per_period[next_index]["PeriodEnd"],
+            "MessageDifference": messages_per_period[next_index]["NumberOfMessages"] - messages_per_period[index]["NumberOfMessages"]
+        })
+
+    log.info(
+        f"writing messages_per_period and  message_difference_per_period json files...")
+    with open(message_difference_output_file_path, mode="w") as f:
+        json.dump(messages_per_period, f)
```

Comment:
I think we you want messages_difference_per_two_firebase_period_output_file_path